apel/db/loader/record_factory.py

```diff
@@ -90,16 +93,27 @@ def create_records(self, msg_text):
                 # crop the string to before the first ':'
                 index = header.index(':')
                 msg_header_type = header[0:index].strip()
-                # msg_header_version = header[index:].strip()
-
-                record_mapping = {
-                    RecordFactory.JR_HEADER: JobRecord,
-                    RecordFactory.SR_HEADER: SummaryRecord,
-                    RecordFactory.NSR_HEADER: NormalisedSummaryRecord,
-                    RecordFactory.SYNC_HEADER: SyncRecord,
-                    RecordFactory.CLOUD_HEADER: CloudRecord,
-                    RecordFactory.CLOUD_SUMMARY_HEADER: CloudSummaryRecord,
-                }
+                msg_header_version = header[index:].strip(': ')
+
+                if msg_header_version == "v0.4":
+                    # This routes us via the new sublcassed v0.4 record objects.
+                    record_mapping = {
+                        RecordFactory.JR_HEADER: JobRecord04,
```

Comment:
Do we need to map JobRecord04 version to JR_04_HEADER something like that. Or is it we are referring to the same header and asking it to refer different classes in both cases?

Reply:
This probably could be neater, but the problem is that we need to accept headers with versions < 0.4 as the original records, and ones with 0.4 as the new type, so we can't simply match on a complete header string.